Rule 1001 for security-hardened-shoot-cluster Panics with nil pointer refference

**What happened**:
Rule 1001 for security-hardened-shoot-cluster panics when `NamespacedCloudProfile` does not specify k8s versions in Spec with:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x104268de0]

goroutine 137 [running]:
github.com/gardener/diki/pkg/provider/garden/ruleset/securityhardenedshoot/rules.(*Rule1001).Run(0x14000125340, {0x1053f7af0, 0x1400084c0a0})
        /Users/I550469/go/src/github.com/gardener/diki/pkg/provider/garden/ruleset/securityhardenedshoot/rules/1001.go:118 +0xd70
github.com/gardener/diki/pkg/shared/ruleset.Run.func1()
        /Users/I550469/go/src/github.com/gardener/diki/pkg/shared/ruleset/ruleset.go:57 +0x178
created by github.com/gardener/diki/pkg/shared/ruleset.Run in goroutine 1
        /Users/I550469/go/src/github.com/gardener/diki/pkg/shared/ruleset/ruleset.go:54 +0x2f4
exit status 2
```

**What you expected to happen**:
Rule to not error

--- a/pkg/provider/garden/ruleset/securityhardenedshoot/rules/1001.go
+++ b/pkg/provider/garden/ruleset/securityhardenedshoot/rules/1001.go
@@ -113,8 +113,10 @@ func (r *Rule1001) Run(ctx context.Context) (rule.RuleResult, error) {
 		return rule.Result(r, rule.ErroredCheckResult("kubernetes version not found in cloudProfile", target)), nil
 	}
 
-	if checkResult, found := r.checkShootVersion(shoot.Spec.Kubernetes.Version, namespacedCloudProfile.Spec.Kubernetes.Versions, target); found {
-		return rule.Result(r, checkResult), nil
+	if namespacedCloudProfile.Spec.Kubernetes != nil {
+		if checkResult, found := r.checkShootVersion(shoot.Spec.Kubernetes.Version, namespacedCloudProfile.Spec.Kubernetes.Versions, target); found {
+			return rule.Result(r, checkResult), nil
+		}
 	}
 	if checkResult, found := r.checkShootVersion(shoot.Spec.Kubernetes.Version, namespacedCloudProfile.Status.CloudProfileSpec.Kubernetes.Versions, target); found {
 		return rule.Result(r, checkResult), nil

--- a/pkg/provider/garden/ruleset/securityhardenedshoot/rules/1001_test.go
+++ b/pkg/provider/garden/ruleset/securityhardenedshoot/rules/1001_test.go
@@ -282,6 +282,30 @@ var _ = Describe("#1001", func() {
 		),
 	)
 
+	It("should correctly use NamespacedCloudProfile status when Spec.Kubernetes is nil", func() {
+		nsCloudProfile.Spec.Kubernetes = nil
+		Expect(fakeClient.Update(ctx, nsCloudProfile)).To(Succeed())
+
+		shoot.Spec.CloudProfile = &gardencorev1beta1.CloudProfileReference{
+			Name: "foo",
+			Kind: "NamespacedCloudProfile",
+		}
+		shoot.Spec.Kubernetes.Version = "2"
+		Expect(fakeClient.Create(ctx, shoot)).To(Succeed())
+
+		r = &rules.Rule1001{
+			Client:         fakeClient,
+			ShootName:      shootName,
+			ShootNamespace: shootNamespace,
+		}
+
+		res, err := r.Run(ctx)
+		Expect(err).To(BeNil())
+		Expect(res).To(Equal(rule.RuleResult{RuleID: ruleID, RuleName: ruleName, Severity: severity, CheckResults: []rule.CheckResult{
+			{Status: rule.Passed, Message: "Shoot uses a Kubernetes version with an allowed classification.", Target: rule.NewTarget("version", "2", "classification", "supported")},
+		}}))
+	})
+
 	Describe("#ValidateOptions", func() {
 		It("should not error when options are correct", func() {
 			options := rules.Options1001{

